### ISSUE TITLE
Security restrictions in GW fail if using empty lists (or non-matchable Resources)

### ISSUE THREAD

**Issue by Alfiva** [universAAL/remote#471]
_Originally Opened: @Alfiva (2015-07-02 11:52:48_)
_Originally Closed: 2015-07-27 13:57:42_

The Multitenant GW requires security restrictions in the &quot;security-definition-file&quot;. There are 2 ttl files in test/resources that could be used as examples, but if these are used, there is an exception:

Exception in thread &quot;Context Bus Strategy Handler&quot; java.lang.ClassCastException:
org.universAAL.middleware.rdf.Resource cannot be cast to org.universAAL.middlew
are.bus.model.matchable.Matchable
        at org.universAAL.ri.gateway.configuration.TurtleFileSecurityDefinition.
getOutgoingMessageOperationChain(TurtleFileSecurityDefinition.java:259)

This is because the examples use empty lists for the matches, and in TurtleFileSecurityDefinition, the &quot;matches&quot; properties are always cast to (Matchable), while an empty list (http://www.w3.org/1999/02/22-rdf-syntax-ns#nil) is a Resource. The code does actually check for Resources and nil, but it always casts to (Matchable) first, thus failing.

It should cast to (Resource) and then the rest of the code will take care of it. This is replicated 4 times, in each &quot;getXXXMessagOperationChain&quot; method of the class.

--

From: _this issue has been automatically imported from our old issue tracker_

**Comment by Alfiva** [comment]
_Original author: @Alfiva (2015-07-27 13:57:42)_

Fixed, now they are not cast until an instanceof check is performed. However this reveals that there are other cast exceptions like this at other levels.

For instance at some point I found that a ServiceCall was attempted to match to the definitions, but a ServiceCall is not a Matchable (and according to Carsten it is not planned to be), so this is a separate bug that must be fixed.
